Internet relay over USB docs; adds requirements.txt

kdmukai opened the pull request:
* Adds instructions for directly configuring the Pi Zero 1.3 by using a connected computer's internet connection relayed over USB. Don't need a second wifi-enabled Pi!
* Adds `docs/` folder to start organizing documentation other than the root README and a place to keep documentation images. Also moved Orange_Pill.JPG into `docs/img`.
* Condensed some of the installation steps from multiple single commands (e.g `sudo apt-get install X`) to a multi-install command (`sudo apt-get install X Y Z`).
* Created typical `requirements.txt` for python dependencies.
* Misc documentation clarification.

newtonick commented:
Great PR! I have a few comments after review of the proposed changes.

> * Adds instructions for directly configuring the Pi Zero 1.3 by using a connected computer's internet connection relayed over USB. Don't need a second wifi-enabled Pi!

Pretty neat about about using relaying internet over USB. I was not aware of this approach. I'm been using a USB ethernet adapter instead of using Wifi. We may want to emphasize that enabling "features" like internet relay should be seen as use for development/experimentation. It makes SeedSigner no longer air gapped. I wouldn't want someone going through the docs and assuming they are required to set this up.

> * Adds `docs/` folder to start organizing documentation other than the root README and a place to keep documentation images. Also moved Orange_Pill.JPG into `docs/img`.

I'm all for moving to a docs folder convention for documentation.

> * Condensed some of the installation steps from multiple single commands (e.g `sudo apt-get install X`) to a multi-install command (`sudo apt-get install X Y Z`).

I'm also good with condensing the OS package installs into a single command from 11 commands.

> * Created typical `requirements.txt` for python dependencies.

Now I've seen the convention of using requirements.txt for python dependencies, but I'm a little confused on why the list of dependencies is so much longer. There was 9 pip package installs the previous way, but the requirements.txt has 28 packages listed. Is this because some of the packages have there own python lib dependencies? I do like that requirements.txt is explicit about dependency versions.



Thanks again for this contribution.

kdmukai commented:
> Pretty neat about about using relaying internet over USB. I was not aware of this approach. I'm been using a USB ethernet adapter instead of using Wifi. We may want to emphasize that enabling "features" like internet relay should be seen as use for development/experimentation. It makes SeedSigner no longer air gapped. I wouldn't want someone going through the docs and assuming they are required to set this up.

Yeah, I'll revise the language around the option and add a note/recommendation to remove the relay at the end for full security.

> Now I've seen the convention of using requirements.txt for python dependencies, but I'm a little confused on why the list of dependencies is so much longer. There was 9 pip package installs the previous way, but the requirements.txt has 28 packages listed. Is this because some of the packages have there own python lib dependencies? I do like that requirements.txt is explicit about dependency versions.

That caught my eye as well. It looks like some of the `apt-get install` steps added a number of python dependencies (I had run a `pip freeze` before doing the `pip install` steps). I didn't realize `apt-get installs` interacted with python in that way. So those were included in the `requirements.txt` that I generated.

I'll do a clean run and this time use a virtualenv so I can isolate just the project-related dependencies. I noticed that there's an earlier pull request (#10) that also generates a requirements.txt. I'd recommend merging that one first, then I'll compare against my results when I update this PR. That submitter deserves contributor credit.

kdmukai commented:
@newtonick I revised the docs per your feedback about the slight security risk created by the internet access over USB option. Also includes a reminder and info on how to remove that access as a final step.

Other changes:
* I generated a clean `requirements.txt` file.
* Removed `sudo` where it was not necessary. Normally `pip` dependencies aren't installed with `sudo` as that basically installs them just for the `su` account. But by installing them with just `pip install` they're just in the `pi` account. So I added `User=pi` to the `systemd` config to force it to use the `pip` dependencies that we just installed to the `pi` user.
* Also changed the `systemd` config to call `main.py` without the absolute path. Since we're setting `WorkingDirectory` the relative call to `main.py` works just fine. There's no effective difference but I think this is (ironically) more explicit to make it clear that we're in `WorkingDirectory`.

SeedSigner commented:
Really appreciate you guys coordinating on this, connectivity via USB was a great thought and thank you for clarifying so users don't inadvertently leave that capability enabled. Also like the idea of going static on the Python libraries we're using, especially for security concerns. All else looks good to me, going to merge.
